src/Lecture1.hs

```diff
@@ -123,4 +142,11 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+lowerAndGreater :: Int -> [Int] -> String
+lowerAndGreater n list =  show n ++ " is greater than " ++ show (checkHead (\x y -> x > y) 0 n list) ++ " elements and lower than " ++ show (checkHead (\x y -> x < y) 0 n list) ++ " elements"
```

Comment:
That's correct! 💯 

However, with this solution, you will walk through the list twice.
You can write it more performant with only one traversal of the list 🙂
It's desirable to write functions that traverse lists single time and consume elements only once. In that case, the composition of such functions will have nice streaming and lazy properties.